tests/test_prompt.py

```diff
@@ -134,3 +135,54 @@ def simple_prompt_func():
     assert isinstance(result, MockResponseFormat)
     assert hasattr(result, "message")
     assert hasattr(result, "confidence")
+
+
+@pytest.mark.test
+def test_clear_cache(prompter: Prompter, tmp_path):
+    """Test that clear_cache removes all cached data."""
+    # Set up temporary cache directory
+    cache_dir = str(tmp_path)
+    os.environ["CURATOR_CACHE_DIR"] = cache_dir
+
+    # Create mock cache files and directories
+    metadata_db_path = os.path.join(cache_dir, "metadata.db")
+    os.makedirs(cache_dir, exist_ok=True)
+
+    # Create a mock metadata.db file
+    with open(metadata_db_path, 'wb') as f:
+        f.write(b'mock db content')
+
+    # Create mock cache directories and files
+    mock_cache_dir = os.path.join(cache_dir, "mock_cache")
+    os.makedirs(mock_cache_dir)
+    with open(os.path.join(mock_cache_dir, "cache.arrow"), 'w') as f:
+        f.write("mock cache data")
+
+    # Create a read-only file to test permission errors
+    readonly_file = os.path.join(cache_dir, "readonly.txt")
+    with open(readonly_file, 'w') as f:
+        f.write("readonly content")
+    os.chmod(readonly_file, 0o444)
+
+    # Verify cache exists
+    assert os.path.exists(metadata_db_path)
+    assert os.path.exists(mock_cache_dir)
+    assert os.path.exists(readonly_file)
+
+    # Clear cache
+    prompter.clear_cache(working_dir=cache_dir)
```

Comment:
for interface, we want `curator.clear_cache()`? that's like a global util thing, not prompter specific.

Reply:
@Devin did you work on this?